main.cpp

```diff
@@ -19,13 +19,23 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
+    CastCallBack(Rewriter& rewriter):_rewriter(rewriter) {
         // Your code goes here
     };
 
     void run(const MatchFinder::MatchResult &Result) override {
         // Your code goes here
+	auto cstmtptr = Result.Nodes.getNodeAs<clang::CStyleCastExpr>("cast");

```

Comment:
What if `cstmtptr` is a null pointer? How to protect the program from crashing in such cases?

Reply:
To protect the program from crashing in such cases we need to add this code after initialisation of cstmtptr:
if (cstmtptr != nullptr) {
        <run the remaining code>
}